raft-tech/TANF-app
# 1755 - As a data analyst, I need to be able to upload SSP data files

## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #1755 #2154

This recording shows a dev belonging to an STT that has SSP access submitting those file:
https://user-images.githubusercontent.com/7119690/200042287-a4cee7ea-080f-4e03-bfad-6bf890d018b1.mov


This recording shows a dev belonging to an STT withou SSP access submitting a TANF file:
https://user-images.githubusercontent.com/7119690/200042478-799a6e90-61a5-415d-9aa1-5771f61b7b8c.mov
 

This recording shows a OFA Admin selecting an STT with SSP access and submitting TANF and SSP files. There is a bug where the previously uploaded file displayed is first file uploaded for that section instead of the last:
https://user-images.githubusercontent.com/7119690/200042661-c9649db7-df46-4588-9134-b08b790ced56.mov


## How to Test

1. 1755
    * Log in as a user with an STT (with `ssp = True`) assigned.
    * Navigate to the Data Files page, a radio button should display allowing the user to select TANF or SSP-MOE data reports. (Not visible if `ssp = False` for the user's STT)
    * Selecting a radio option should apply to the search (`GET` request)
    * When search is clicked, the Reports header is uploaded to include the submission type (TANF or SSP-MOE)
    * When a file is uploaded, if SSP-MOE is selected from the radio buttons, `Ssp` will be prepended to the data report's `section` in the Django Admin Console (when TANF is selected, nothing is prepended)
    * When the radio button selection is changed, search must be clicked before the selection will apply to uploaded files. If a file is attached (but not submitted) and search selections are changed, a modal will appear when clicking search (#2209 frontend changes)
    * (Revalidate) Selecting a radio option should apply to the search (`GET` request)
       * __NOTE__: the "Download File" button does not download the latest version of the file if there are multiple submitted. I believe this is an existing bug.
2. 2154
   * When a file is submitted for an STT with `ssp = True`, the file name should be selected from the `SSP ${section}` of the STT's `filenames`.
       * __NOTE__: if the section doesn't exist in the STT's `filenames`, this will throw an exception. I believe this is an existing issue and will be addressed with future work.


## Deliverables

1755 ACS:
- [x] STT Data Analyst can select the type of data report being uploaded
- [x] Radio button section not visible to users of STTs without SSP indicator
- [x] Radio select defaults to TANF
- [x] Heading 1 is modified to read "Data Files" instead of "TANF Data Files"
- [x] Heading 2 prints radio selection between [STT Name] and [Fiscal Year]

2154 ACS:
- [x] SSP section is available in data file upload in frontend
- [x] SSP section files are correctly renamed as per naming convention in STT model filename
- [x] filename property in datafile model has the correct logic to rename STT section files using the [states table](https://github.com/raft-tech/TANF-app/blob/develop/tdrs-backend/tdpservice/stts/management/commands/data/states.csv)


### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
